### PR TITLE
Update prepare_data() to better use the passed field

### PR DESCRIPTION
**Pull request by mboynes**

### CODE REVIEW

**Comment by bcampeau** [comment]
:horse:

File: php/context/class-fieldmanager-context.php [hunk 1]
```diff
@@ -59,11 +59,11 @@ protected function prepare_data( $old_value = null, $new_value = null, $fm = nul
 			$fm = $this->fm;
 		}
 		if ( null === $new_value ) {
-			$new_value = isset( $_POST[ $this->fm->name ] ) ? $_POST[ $this->fm->name ] : '';
```
Comment (danielbachhuber) [inline php/context/class-fieldmanager-context.php]: Let's revert this change for now, because we aren't fully certain it doesn't have unintended consequences.
